site/docs-md/guides/push-notifications-firebase.md

```diff
@@ -0,0 +1,369 @@
+
+```typescript
+import { Component, OnInit } from '@angular/core';
+
+import {
+  Plugins,
+  PushNotification,
+  PushNotificationToken } from '@capacitor/core';
+
+const { PushNotifications } = Plugins;
+
+@Component({
+  selector: 'app-home',
+  templateUrl: 'home.page.html',
+  styleUrls: ['home.page.scss'],
+})
+
+export class HomePage implements OnInit {
+
+  ngOnInit() {
+    console.log('Initializing HomePage');
+
+    PushNotifications.register();
+
+    PushNotifications.addListener('registration', 
+      (token: PushNotificationToken) => {
+        alert('Push registration success, token: ' + token.value);
+      }
+    );
+
+    PushNotifications.addListener('registrationError', 
+      (error: any) => {
+        alert('Error on registration: ' + JSON.stringify(error));
+      }
+    );
+
+    PushNotifications.addListener('pushNotificationReceived', 
+      (notification: PushNotification) => {
+        alert('Push received: ' + JSON.stringify(notification));
+      }
+    );
+}
+```
+
+## Creating a Project for your App on Firebase
+
+Before we can connect Firebase Cloud Messaging to your application and send push notifications, you'll need to start a project in Firebase.
+
+Go to the [Firebase Console](https://console.firebase.google.com/) and click the **Add project** button.
+
+Name the project, accept the Firebase ToS and click **Create project** to continue. A Project ID should be automatically generated for you.
+
+## Android
+
+### Integrating Firebase with the Android app
+
+This section more-or-less mirrors the [setting up Firebase using the Firebase console documentation](https://firebase.google.com/docs/android/setup?authuser=0). See below for specific Capacitor-related notes.
+
+Go to the Project Overview page for your Firebase project and, under the **Grow** section, click the **Cloud Messaging** option. At the top, click on the **Android** icon.
+
+The next screen will ask you for some information about your application.
+
+- Your **Android package name** should match the **appId** from your `capacitor.config.json` file
+- We used `com.mydomain.myappname` for this Capacitor app ID, so that is what we'll use for this entry.
+- Nickname and Debug Signing Certificate are optional
+
+Then click the **Register app** button.
+
+### Download and Use the `google-services.json` file
+
+The next prompt will ask you to download a `google-services.json` file. This file contains the information your Capacitor app needs to connect to Firebase from Android.
+
+Download the `google-services.json` file to your local machine. Then move the file into your Capacitor Android project directory, specifically under `android/app/`.
+
+![Google Services JSON Location for Android](/assets/img/docs/guides/firebase-push-notifications/google-services-location-android.png)
+
+We don't need to *add* any permissions to our project because Capacitor projects automatically include a version of `firebase-messaging` in our `app/build.gradle` file:
```

Comment:
push don't need permissions, also, you are talking about dependencies, not permissions.
Not sure if it's good to add this with versions as I'm working right now on updating to latest version (18.0.0), which will make this out of date shortly after being merged.